textile/features.textile

```diff
@@ -616,6 +617,7 @@ h3(#realtime-channel). RealtimeChannel
 *** @(RTL4k1)@ If any channel parameters are requested (which may be through the @params@ field of the @ATTACH@ message or some other way opaque to the client library), the @ATTACHED@ (and any subsequent @ATTACHED@ s) will include a @params@ property (also a @Dict<String, String>@) containing the subset of those params that the server has recognised and validated. This should be exposed as a read-only @params@ field of the @RealtimeChannel@ (or a @getParams()@ method where that is more idiomatic). An @ATTACHED@ message with no @params@ property must be treated as equivalent to a @params@ of @{}@ (that is, @RealtimeChannel.params@ should be set to the empty dict)
 ** @(RTL4l)@ If the user has specified a @modes@ array in the @ChannelOptions@ ("@TB2d@":#TB2d), it must be encoded as a bitfield per "@TR3@":#TR3 and set as the @flags@ field of the @ATTACH@ @ProtocolMessage@. (For the avoidance of doubt, when multiple different spec items require flags to be set in the @ATTACH@, the final @flags@ field should be the bitwise OR of them all)
 ** @(RTL4m)@ On receipt of an @ATTACHED@, the client library should decode the @flags@ into an array of @ChannelMode@ s (that is, the same format as @ChannelOptions.modes@) and expose it as a read-only @modes@ field of the @RealtimeChannel@ (or a @getModes()@ method where that is more idiomatic). This should only contain @ChannelMode@ s: it should not contain flags which are not modes (see "@TB2d@":#TB2d)
+** @(RTL4n)@ Optionally, upon success, may return the @ChannelStateChange@ object once the channel is attached. If the channel is already attached, a synthetic @ChannelStateChange@ from @ATTACHED@ to @ATTACHED@ may be returned.
```

Comment:
`RTL2g` says that "The library must never emit a `ChannelState` `ChannelEvent` for a state equal to the previous state". I think our options are to either emit an `UPDATE` `ChannelEvent` or to return `null`.

Reply:
I don't think RTL2g applies here, it's meant for the `RealtimeChannel` `EventEmitter` and makes sense there because it's helpful to be able to assume that the listener in `channel.on('attached', listener)` only fires when the channel actually becomes attached. In this case i don't think the same reasoning applies. Also, the `UPDATE` channel event is just the name of the event which comes with a `ChannelStateChange`. Since we're not emitting an event here, that's not an option.

Having thought a bit about this I will change it to return `null` though, mainly just to not mislead users into thinking that the channel state has changed just because they have a `ChannelStateChange` object